src/main/java/application/FormplayerAuthFilter.java

```diff
@@ -38,26 +55,66 @@ public void init(FilterConfig filterConfig) throws ServletException {
     @Override
     public void doFilter(ServletRequest req, ServletResponse res,
                          FilterChain chain) throws IOException, ServletException {
-        HttpServletRequest request = (HttpServletRequest) req;
+        FormplayerHttpRequest request = new FormplayerHttpRequest((HttpServletRequest) req);
         if (isAuthorizationRequired(request)) {
-            if (!authorizeRequest(request)) {
+            // These are order dependent
+            if (getSessionId(request) == null) {
+                setResponseUnauthorized((HttpServletResponse) res);
+                return;
+            }
+            setToken(request);
```

Comment:
I think it might be cleaner to pass in the repositories to the FormplayerHttpRequest class and then do all this instantiation work there